packages/material-ui/src/Breadcrumbs/Breadcrumbs.js

```diff
@@ -26,99 +26,100 @@ const styles = {
   separator: {},
 };
 
-class Breadcrumbs extends React.Component {
-  state = {
-    expanded: false,
-  };
-
-  handleClickExpand = () => {
-    this.setState({ expanded: true });
-  };
+function insertSeparators(items, className, separator) {
+  return items.reduce((acc, current, index) => {
+    if (index < items.length - 1) {
+      acc = acc.concat(
+        current,
+        <BreadcrumbSeparator
+          // eslint-disable-next-line react/no-array-index-key
+          key={`separator-${index}`}
+          className={className}
+        >
+          {separator}
+        </BreadcrumbSeparator>,
+      );
+    } else {
+      acc.push(current);
+    }
 
-  insertSeparators(items) {
-    return items.reduce((acc, current, index) => {
-      if (index < items.length - 1) {
-        acc = acc.concat(
-          current,
-          <BreadcrumbSeparator
-            // eslint-disable-next-line react/no-array-index-key
-            key={`separator-${index}`}
-            className={this.props.classes.separator}
-          >
-            {this.props.separator}
-          </BreadcrumbSeparator>,
-        );
-      } else {
-        acc.push(current);
-      }
+    return acc;
+  }, []);
+}
 
-      return acc;
-    }, []);
-  }
+const Breadcrumbs = React.forwardRef(function Breadcrumbs(props, ref) {
+  const {
+    children,
+    classes,
+    className,
+    component: Component,
+    itemsAfterCollapse,
+    itemsBeforeCollapse,
+    maxItems,
+    separator,
+    ...other
+  } = props;
 
-  renderItemsBeforeAndAfter(allItems) {
-    const { maxItems, itemsBeforeCollapse, itemsAfterCollapse } = this.props;
+  const [expanded, setExpanded] = React.useState(false);
 
-    // This defends against someone passing weird input, to ensure that if all
-    // items would be shown anyway, we just show all items without the EllipsisItem
-    if (itemsBeforeCollapse + itemsAfterCollapse >= allItems.length) {
-      warning(
-        false,
-        [
-          'Material-UI: you have provided an invalid combination of properties to the Breadcrumbs.',
-          // eslint-disable-next-line max-len
-          `itemsAfterCollapse={${itemsAfterCollapse}} + itemsBeforeCollapse={${itemsBeforeCollapse}} >= maxItems={${maxItems}}`,
-        ].join('\n'),
-      );
-      return allItems;
-    }
+  const handleClickExpand = () => {
```

Comment:
I think that we can add useCallback, it would help with the leading React.useMemo logic.